source/macos/AdaptiveCards/AdaptiveCards/Views/ACRDateField.swift

```diff
@@ -18,29 +18,24 @@ class ACRDateField: NSView, InputHandlingViewProtocol {
         return formatter
     }()
     
-    private (set) lazy var textField: NSTextField = {
-        let view = NSTextField()
+    private (set) lazy var textField: ACRTextField = {
+        let view = ACRTextField(dateTimeFieldWith: config)
         view.translatesAutoresizingMaskIntoConstraints = false
         view.isEditable = true
         view.isSelectable = false
-        view.isBordered = true
+        view.cell?.lineBreakMode = .byTruncatingTail
+        view.stringValue = ""
+        view.textFieldDelegate = self
        return view
     }()
-    
-    private (set) lazy var clearButton: NSButtonWithImageSpacing = {
-        let resourceName = isDarkMode ? "clear_18_w" : "clear_18"
-        let view = NSButtonWithImageSpacing(image: BundleUtils.getImage(resourceName, ofType: "png") ?? NSImage(), target: self, action: #selector(handleClearAction))
-        view.translatesAutoresizingMaskIntoConstraints = false
-        view.wantsLayer = true
-        view.layer?.backgroundColor = NSColor.clear.cgColor
-        view.isBordered = false
-        return view
-    }()
 
     private lazy var iconButton: NSButtonWithImageSpacing = {
         let calendarResourceName = isDarkMode ? "calendar-month-dark" : "calendar-month-light"
         let clockResourceName = isDarkMode ? "recents_20_w" : "recents_20"
-        let view = NSButtonWithImageSpacing(image: (isTimeMode ? BundleUtils.getImage(clockResourceName, ofType: "png") ?? NSImage() : BundleUtils.getImage(calendarResourceName, ofType: "png") ?? NSImage()), target: self, action: #selector(mouseDown(with:)))
+        let calendarImage = BundleUtils.getImage(calendarResourceName, ofType: "png")
+        let clockImage = BundleUtils.getImage(clockResourceName, ofType: "png")
+        let inputFieldConfig = config.inputFieldConfig
+        let view = NSButtonWithImageSpacing(image: (isTimeMode ? inputFieldConfig.clockImage ?? clockImage : inputFieldConfig.calendarImage ?? calendarImage) ?? NSImage(), target: self, action: #selector(mouseDown(with:)))
```

Comment:
```suggestion
        let inputFieldConfig = config.inputFieldConfig
        let view = NSButtonWithImageSpacing(image: (isTimeMode ? inputFieldConfig.clockImage ?? clockImage : inputFieldConfig.calendarImage ?? calendarImage) ?? NSImage(), target: self, action: #selector(handleOpenPickerAction))
```
It's better to have separate method instead of using system methods like `mouseDown`.